fuel-core/src/database.rs

```diff
@@ -339,11 +367,7 @@ impl InterpreterStorage for Database {
                 // FIXME: Get producer address from block signature.
                 // block_id -> Signature
                 let signature = Signature::default();
-                let message = unsafe {
-                    fuel_core_interfaces::common::fuel_crypto::Message::from_bytes_unchecked(
-                    block.header.id().into(),
-                )
-                };
+                let message = block.header.id().into_message();
                 // TODO: throw an error if public key isn't recoverable
                 //  when implementing signing (https://github.com/FuelLabs/fuel-core/issues/668)
                 let public_key = signature.recover(&message).unwrap_or_default();
```

Comment:
This should not be `unwrap_or_default` anymore.

Reply:
fixed to use the correct signature and public key